lib_xua/doc/rst/sw_dfu.rst

```diff
@@ -3,25 +3,117 @@
 Device Firmware Upgrade (DFU)
 =============================
 
-The DFU interface handles updates to the boot image of the device. The DFU code is called from 
-the Endpoint 0 core.
+The DFU implementation in lib_xua is compliant with version 1.1 of
+`Universal Serial Bus Device Class Specification for Device Firmware Upgrade <https://www.usb.org/sites/default/files/DFU_1.1.pdf>`_.
+The device descriptors expose a DFU interface that handles updates to the boot image of the device over USB.
 
-The interface links USB to the XMOS flash user library (see :ref:`libflash_api`). In Application 
-mode the DFU can accept commands to reset the device into DFU mode. There are two ways to do this:
 
--  The host can send a ``DETACH`` request and then reset the
-   device. If the device is reset by the host within a specified
-   timeout, it will start in DFU mode (this is initially set to
-   one second and is configurable from the host).
+On receiving DFU commands from the host, the ``DFUDeviceRequests`` function is called from the Endpoint 0 core.
+This function calls the DFU handler functions over the ``dfuInterface`` XC interface.
+The DFU handler thread, ``DFUHandler`` that implements the server side of the ``dfuInterface`` has to be
+scheduled on the same tile as the flash so it can access the flash memory.
+The ``dfuInterface`` interface links USB to the XMOS flash user library (see :ref:`libflash_api`).
 
--  The host can send a custom user request
-   ``XMOS_DFU_RESETDEVICE`` to the DFU interface that 
-   resets the device immediately into DFU mode.
+
+The DFU interface is enabled by default (See XUA_DFU_EN define in ``xua_conf_default.h``).
+When DFU is enabled, there are two sets of descriptors that the device can export, depending on the mode in which it operates.
+There's a descriptor set for the runtime mode, which is the mode the device normally operates in and a set of descriptors for the DFU mode.
+
+In the runtime mode, the DFU interface is one of potentially multiple interfaces that the device exposes. :ref:`dfu_interface_runtime` shows the DFU interface
+descriptors when enumerating in runtime mode as seen in a `Beagle USB analyser <https://www.totalphase.com/products/data-center/>`_ trace:
+
+ .. _dfu_interface_runtime:
+
+ .. figure:: images/dfu_interface_runtime_mode.png
+   :width: 100%
+
+   DFU interface when part of runtime mode descriptor set
+
+Note the **bInterfaceProtocol** field set to **Runtime**.
+
+In DFU mode, the device exports the DFU descriptor set. The DFU mode descriptors specify only one interface, the DFU interface.
+:ref:`dfu_interface_dfu` shows the DFU interface
+descriptors when enumerating in DFU mode as seen in a Beagle USB analyser trace:
```

Comment:
```suggestion
descriptors when enumerating in DFU mode as seen in a Beagle USB analyser trace.
```

because the diagram doesn't render underneath this line.